### PR TITLE
feat(kubernetes): create, delete, list, show & versions

### PR DESCRIPTION
**Pull request by villevsv-upcloud**

### CODE REVIEW

File: internal/commands/kubernetes/delete.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func (s *deleteCommand) Execute(exec commands.Executor, arg string) (output.Output, error) {
+	svc := exec.All()
+	msg := fmt.Sprintf("Deleting Kubernetes cluster %v", arg)
+	exec.PushProgressStarted(msg)
+
+	err := svc.DeleteKubernetesCluster(&request.DeleteKubernetesClusterRequest{
+		UUID: arg,
+	})
+	if err != nil {
+		return commands.HandleError(exec, fmt.Sprintf("%s: failed", msg), err)
```
Comment (kangasta) [inline internal/commands/kubernetes/delete.go:42]: Other `HandleError` calls do not add `:  failed` to the end of the status message anymore as failure is indicated with `✗` before the message.

File: internal/commands/kubernetes/show.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+								{Title: "Zone", Value: cluster.Zone},
+								{Title: "Operational state:", Value: cluster.State, Format: format.KubernetesState},
+							},
+						},
+					},
+				},
+			},
+			output.CombinedSection{
+				Title: "Node groups:",
+				Contents: output.Table{
```
Comment (kangasta) [inline internal/commands/kubernetes/show.go:99]: This table will be quite wide. Would it make sense to list only `name`, `count`,  `plan` and `storage` here and have `upctl kubernetes node-group` sub-command with `list` and `show` commands? Not sure if we have anything similar for other resources currently, though.

File: internal/commands/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	}
+}
+
+type kubernetesCommand struct {
+	*commands.BaseCommand
+}
+
+// InitCommand implements Command.InitCommand
+func (k *kubernetesCommand) InitCommand() {
+	k.Cobra().Aliases = []string{"uks"}
```
Comment (kangasta) [inline internal/commands/kubernetes/kubernetes.go:20]: Could also have `k8s` as alias

File: internal/commands/kubernetes/show.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		}
+		taints := make([]string, 0)
+		for _, v := range nodeGroup.Taints {
+			taints = append(taints, fmt.Sprintf("Key: %s\nValue: %s\nEffect: %s", v.Key, v.Value, v.Effect))
+		}
+		nodeGroupRows = append(nodeGroupRows, output.TableRow{
+			nodeGroup.Name,
+			nodeGroup.Count,
+			nodeGroup.Plan,
+			nodeGroup.Storage,
```
Comment (kangasta) [inline internal/commands/kubernetes/show.go:68]: I think we have helper function that could be used to display storage name instead of UUID in human readable output, but that could be introduced here also in separate PR.

**Comment by kangasta** [comment]
Didn't yet have time to go through all of the files, but added few comments already

File: internal/commands/kubernetes/kubernetes.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package kubernetes
+
+import (
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/commands"
+)
+
+// BaseKubernetesCommand creates the base "kubernetes" command
+func BaseKubernetesCommand() commands.Command {
+	return &kubernetesCommand{
+		commands.New("kubernetes", "Manage Kubernetes clusters"),
```
Comment (kangasta) [inline internal/commands/kubernetes/kubernetes.go:10]: Should we add some sort of experimental note here? E.g., our Terraform provider docs have: `NOTE: this is an experimental feature in an alpha phase, the resource definition will change in the future.`.

File: internal/completion/kubernetes.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package completion
+
+import (
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/service"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/request"
+	"github.com/spf13/cobra"
+)
+
+// Kubernetes implements argument completion for load balancers, by uuid or name.
```
Comment (kangasta) [inline internal/completion/kubernetes.go:10]: `for load balancers` → `for kubernetes clusters`

File: internal/commands/kubernetes/kubernetes.go (hunk 5 above)
Comment (kangasta) [reply internal/commands/kubernetes/kubernetes.go:10]: Go SDK seems to use `(EXPERIMENTAL)` at the end of the doc comment

File: internal/completion/kubernetes.go (hunk 6 above)
Comment (villevsv-upcloud) [reply internal/completion/kubernetes.go]: Fixed in https://github.com/UpCloudLtd/upcloud-cli/pull/190/commits/5d5ce9c85dcfbba240bd47a1f1d887e5f2679bbd

File: internal/commands/kubernetes/delete.go (hunk 1 above)
Comment (villevsv-upcloud) [reply internal/commands/kubernetes/delete.go]: Fixed in https://github.com/UpCloudLtd/upcloud-cli/pull/190/commits/2e477343c90c637e44300d4a1d4cb81c7453fd03

File: internal/commands/kubernetes/kubernetes.go (hunk 5 above)
Comment (villevsv-upcloud) [reply internal/commands/kubernetes/kubernetes.go]: Fixed in https://github.com/UpCloudLtd/upcloud-cli/pull/190/commits/2be93a4a7d71bdec90e718c80c5c0842082ab6c3

File: internal/commands/kubernetes/show.go (hunk 2 above)
Comment (kangasta) [reply internal/commands/kubernetes/show.go:99]: And as this only affects human output and is output of experimental command, I think we can also tune the output in future PR.

File: internal/commands/kubernetes/versions.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/commands"
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/output"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/request"
+)
+
+// VersionsCommand creates the "kubernetes versions" command
+func VersionsCommand() commands.Command {
+	return &versionsCommand{
+		BaseCommand: commands.New("versions", "List available versions for Kubernetes node groups", "upctl kubernetes versions"),
```
Comment (kangasta) [inline internal/commands/kubernetes/versions.go:13]: Will the version parameter be for cluster or node-group?

Comment (villevsv-upcloud) [reply internal/commands/kubernetes/versions.go:13]: You are correct! It is indeed the cluster version, will fix this.

File: internal/commands/kubernetes/create.go [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+				--network 03e5ca07-f36c-4957-a676-e001e40441eb \
+				--network-cidr "172.16.1.0/24" \
+				--node-group count=2,name=my-minimal-node-group,plan=K8S-2xCPU-4GB, \
+				--zone de-fra1`,
+			`upctl kubernetes create \
+				--name my-cluster \
+				--network 03e5ca07-f36c-4957-a676-e001e40441eb \
+				--network-cidr "172.16.1.0/24" \
+				--node-group \
+					count=4,kubelet-arg="log-flush-frequency=5s",label="owner=devteam",label="env=dev",name=my-node-group,plan=K8S-4xCPU-8GB,ssh-key="ssh-ed25519 AAAAo admin@user.com",ssh-key="/path/to/your/public/ssh/key.pub",storage=01000000-0000-4000-8000-000160010100,taint="env=dev:NoSchedule",taint="env=dev2:NoSchedule" \
```
Comment (kangasta) [inline internal/commands/kubernetes/create.go:35]: I wonder how this long example line will render in `--help` output and docs. Also something that we could polish closer to the beta release, though.

File: internal/commands/kubernetes/create.go [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+// ExecuteWithoutArguments implements commands.NoArgumentCommand
+func (c *createCommand) ExecuteWithoutArguments(exec commands.Executor) (output.Output, error) {
+	svc := exec.All()
+	msg := fmt.Sprintf("Creating cluster %s", c.params.Name)
+	exec.PushProgressStarted(msg)
+	if err := c.params.processParams(); err != nil {
+		return nil, err
+	}
+
+	exec.Debug("sending r", "params", c.params)
```
Comment (kangasta) [inline internal/commands/kubernetes/create.go:235]: Could make this more descriptive (e.g. `Creating kubernetes cluster, params:`) or remove

File: internal/commands/kubernetes/show.go (hunk 2 above)
Comment (villevsv-upcloud) [reply internal/commands/kubernetes/show.go]: Fixed to a narrower version, example: https://github.com/UpCloudLtd/upcloud-cli/blob/feat/kubernetes/internal/commands/kubernetes/show_test.go#L99

File: internal/commands/kubernetes/create.go [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+}
+
+// InitCommand implements Command.InitCommand
+func (c *createCommand) InitCommand() {
+	fs := &pflag.FlagSet{}
+	c.params = createParams{CreateKubernetesClusterRequest: request.CreateKubernetesClusterRequest{}}
+
+	fs.StringVar(&c.params.Name, "name", "", "Kubernetes cluster name.")
+	fs.StringVar(&c.params.Network, "network", "", "Network to use. The value should be UUID of a private network.")
+	fs.StringVar(&c.params.NetworkCIDR, "network-cidr", "", "CIDR of the network being used.")
```
[the remark this answers is not in the document]
Comment (villevsv-upcloud) [reply internal/commands/kubernetes/create.go]: Fixed

File: internal/commands/kubernetes/create.go (hunk 9 above)
Comment (villevsv-upcloud) [reply internal/commands/kubernetes/create.go]: Removed

File: internal/commands/kubernetes/kubernetes.go (hunk 3 above)
Comment (villevsv-upcloud) [reply internal/commands/kubernetes/kubernetes.go]: Added

File: internal/commands/kubernetes/versions.go (hunk 7 above)
Comment (villevsv-upcloud) [reply internal/commands/kubernetes/versions.go]: Fixed
